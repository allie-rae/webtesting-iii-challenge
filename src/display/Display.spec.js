// Test away!
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'

import Display from './Display';

test("Display renders without crashing", () => {
    const display = render(<Display />);
    expect(display.asFragment()).toMatchSnapshot();
})


test("Displays Open / Unlocked", () => {
    const display = render(<Display closed={false} locked={false} />);

    expect(display.getByTestId('lockedDisplay')).toBeInTheDocument();
    expect(display.getByTestId('closedDisplay')).toBeInTheDocument();

})

test("Displays locked & unlocked / red & green Classes", () => {
    const display = render(<Display closed={false} locked={false} />);

    const lockedDisplay = display.getByText(/unlocked/i);

    lockedDisplay.classList.contains('green-led') && expect(lockedDisplay.textContent).toMatch(/unlocked/i);
    lockedDisplay.classList.contains('red-led') && expect(lockedDisplay.textContent).toMatch(/locked/i);
})

test("Displays locked and red class", () => {
    const display = render(<Display closed={true} locked={true} />);

    expect(display.getByText(/locked/i));
    expect(closedDisplay.classList.contains('red-led'));

})
