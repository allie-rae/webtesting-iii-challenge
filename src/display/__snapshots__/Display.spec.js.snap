// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Snapshot 1`] = `
<DocumentFragment>
  <div
    class="display panel"
  >
    <div
      class="led green-led"
      data-testid="lockedDisplay"
    >
      Unlocked
    </div>
    <div
      class="led green-led"
      data-testid="closedDisplay"
    >
      Open
    </div>
  </div>
</DocumentFragment>
`;

exports[`Display renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="display panel"
  >
    <div
      class="led green-led"
      data-testid="lockedDisplay"
    >
      Unlocked
    </div>
    <div
      class="led green-led"
      data-testid="closedDisplay"
    >
      Open
    </div>
  </div>
</DocumentFragment>
`;
